lib/src/layouts/yaru_master_detail_page.dart

```diff
@@ -88,36 +91,40 @@ class YaruMasterDetailPage extends StatefulWidget {
   final ValueChanged<int?>? onSelected;
 
   /// An optional controller that can be used to navigate to a specific index.
-  final ValueNotifier<int>? controller;
+  final YaruPageController? controller;
 
   @override
   _YaruMasterDetailPageState createState() => _YaruMasterDetailPageState();
 }
 
 class _YaruMasterDetailPageState extends State<YaruMasterDetailPage> {
-  var _index = -1;
-  var _previousIndex = 0;
-
   double? _previousPaneWidth;
+  late final YaruPageController _controller;
+  late final int _length;
 
-  void _setIndex(int index) {
-    _previousIndex = _index;
-    _index = index;
-    widget.onSelected?.call(index == -1 ? null : index);
-  }
+  void _updateController() => _controller = widget.controller ??
+      YaruPageController(
+        length: _length,
+        initialIndex: widget.initialIndex ?? -1,
+      );
 
   @override
   void initState() {
     super.initState();
-    _index = widget.initialIndex ?? -1;
+    _length = widget.length ?? widget.controller!.length;
```

Comment:
re-sync in `didUpdateWidget()` or make it a convenience getter instead

Reply:
Using a getter is a good idea - is there any reason this wouldn't work in the compact layout as well?

Reply:
i like the getter too and i _thought_ i had proposed it above for CL. i don't know what happened but sorry for the confusion - i kept mixing up the classes while reviewing